src/Microsoft.Android.Runtime.NativeAOT/Android.Runtime.NativeAOT/DiagnosticSettings.cs

```diff
@@ -0,0 +1,136 @@
+using System.IO;
+using System.Text;
+using System.Runtime.InteropServices;
+
+using Java.Interop;
+
+namespace Microsoft.Android.Runtime;
+
+struct DiagnosticSettings {
+
+	public  bool    LogJniLocalReferences;
+	private string? LrefPath;
+
+	public  bool    LogJniGlobalReferences;
+	private string? GrefPath;
+
+	private TextWriter?     GrefLrefLog;
+
+
+	public  TextWriter? GrefLog {
+		get {
+			if (!LogJniGlobalReferences) {
+				return null;
+			}
+			return ((LrefPath != null && LrefPath == GrefPath)
+					? GrefLrefLog ??= CreateWriter (LrefPath)
+					: null)
+				??
+				((GrefPath != null)
+					? CreateWriter (GrefPath)
+					: null)
+				??
+				new LogcatTextWriter (AndroidLogLevel.Debug, "NativeAot:GREF");
+		}
+	}
+
+	public  TextWriter? LrefLog {
+		get {
+			if (!LogJniLocalReferences) {
+				return null;
+			}
+			return ((LrefPath != null && LrefPath == GrefPath)
+					? GrefLrefLog ??= CreateWriter (LrefPath)
+					: null)
+				??
+				((LrefPath != null)
+					? CreateWriter (LrefPath)
+					: null)
+				??
+				new LogcatTextWriter (AndroidLogLevel.Debug, "NativeAot:LREF");
+		}
+	}
```

Comment:
I read through these twice, as it seemed easy to have a typo. I think they are right.